docs/stacks-guide-rook.md

```diff
@@ -0,0 +1,830 @@
+created:
+
+*Command*
+```bash
+kubectl get kubernetesapplication -n yugastore-app
+```
+*Output*
+```bash
+NAME            CLUSTER         STATUS      DESIRED   SUBMITTED
+rook-yugabyte   yugastore-k8s   Submitted   7         7
+```
+
+You can also view the individual `KubernetesApplicationResources` as they are
+created:
+
+*Command*
+```bash
+kubectl get kubernetesapplicationresources -n yugastore-app
+```
+*Output*
+```bash
+rook-app-namespace    Namespace                  rook-yugabytedb                 yugastore-k8s   Submitted
+rook-clusterrole      ClusterRole                rook-yugabytedb-operator        yugastore-k8s   Submitted
+rook-crds             CustomResourceDefinition   ybclusters.yugabytedb.rook.io   yugastore-k8s   Submitted
+rook-namespace        Namespace                  rook-yugabytedb-system          yugastore-k8s   Submitted
+rook-serviceaccount   Deployment                 rook-yugabytedb-operator        yugastore-k8s   Submitted
+```
+
+It is not unusual for the `rook-yugabytedb-operator` to fail on first attempt.
+It is reliant on the `ClusterRole` and will fail if it is not present when the
+container starts. However, the Crossplane `workload` controllers will take care
+of recreating it for us!
+
+## Deploy Yugastore alongside YugabyteDB
+
+Now that we have a GKE cluster up and running, we can create our YugabyteDB
+cluster and install Yugastore alongside it.
+
+```bash
+cat > yugastore.yaml <<EOF
+apiVersion: database.crossplane.io/v1alpha1
+kind: PostgreSQLInstance
+metadata:
+  name: yugastore-db
+  namespace: yugastore-app
+  labels:
+    app: yugastore
+spec:
+  classSelector:
+    matchLabels:
+      app: yugastore
+  writeConnectionSecretToRef:
+    name: yugastore-db-secret
+---
+apiVersion: workload.crossplane.io/v1alpha1
+kind: KubernetesApplication
+metadata:
+  name: yugastore
+  namespace: yugastore-app
+  labels:
+    app: yugastore
+spec:
+  resourceSelector:
+    matchLabels:
+      app: yugastore
+  clusterSelector:
+    matchLabels:
+      app: yugastore
+  resourceTemplates:
+  - metadata:
+      name: yugastore-namespace
+      labels:
+        app: yugastore
+    spec:
+      template:
+        apiVersion: v1
+        kind: Namespace
+        metadata:
+          name: rook-yugastore
+          labels:
+            app: yugastore
+  - metadata:
+      name: yugastore-deployment
+      labels:
+        app: yugastore
+    spec:
+      template:
+        apiVersion: apps/v1
+        kind: Deployment
+        metadata:
+          namespace: rook-yugastore
+          name: yugastore
+          labels:
+            app: yugastore
+        spec:
+          selector:
+            matchLabels:
+              app: yugastore
+          template:
+            metadata:
+              labels:
+                app: yugastore
+            spec:
+              containers:
+                - name: yugastore
+                  image: gcr.io/crossplane-playground/yugastore:latest
```

Comment:
My PR on [Yugastore](https://github.com/yugabyte/yugastore) has not been merged yet so I am hosting the image here for now. It is probably fine for us to host the image, but likely want to put in docker hub or a more official gcr project other than `crossplane-playground`.

Reply:
It's merged! https://github.com/yugabyte/yugastore/pull/4

Reply:
@prasek yep! However, the app is still not ready with that merge unfortunately. Going to see if I can get it ironed out this weekend, but we would also have to get them to build an image and publish, and would be subject to any changes, so may just be best to host it ourselves.